contracts/src/SuccinctVApp.sol

```diff
@@ -390,54 +341,77 @@ contract SuccinctVApp is
-                continue;
-            }
+        // Ensure that the receipt is consistent with the transaction.
+        Receipts.assertEq(transactions[onchainTxId], _receipt);
 
-            // If the transaction completed, run a handler for the transaction.
-            if (variant == TransactionVariant.Deposit) {
-                // No-op.
-            } else if (variant == TransactionVariant.Withdraw) {
-                Withdraw memory withdraw = abi.decode(_publicValues.receipts[i].action, (Withdraw));
-                _processWithdraw(withdraw.account, withdraw.amount);
-            } else if (variant == TransactionVariant.CreateProver) {
-                // No-op.
-            } else {
-                revert TransactionVariantInvalid();
-            }
+        // Update the transaction status.
+        transactions[onchainTxId].status = _receipt.status;
+
+        // If the transaction failed, emit the revert event and skip the rest of the loop.
+        if (_receipt.status == TransactionStatus.Reverted) {
+            emit TransactionReverted(onchainTxId, _receipt.variant, _receipt.action);
+            return;
+        }
 
-            // Emit the completed event.
-            emit TransactionCompleted(onchainTxId, variant, _publicValues.receipts[i].action);
+        // If the transaction completed, run a handler for the transaction.
+        if (_receipt.variant == TransactionVariant.Deposit) {
+            // No-op.
+        } else if (_receipt.variant == TransactionVariant.CreateProver) {
+            // No-op.
+        } else {
+            revert TransactionVariantInvalid();
         }
+
+        // Emit the completed event.
+        emit TransactionCompleted(onchainTxId, _receipt.variant, _receipt.action);
     }
 
-    /// @dev Processes a withdrawal by creating a claim for the amount.
-    function _processWithdraw(address _account, uint256 _amount) internal {
-        // Update the state.
-        claimableWithdrawal[_account] += _amount;
+    function _handleOffchainReceipt(Receipt memory _receipt) internal {
+        // Ensure that the receipt has of the expected statuses.
+        if (
+            _receipt.status == TransactionStatus.None
+                || _receipt.status == TransactionStatus.Pending
+        ) {
+            revert ReceiptStatusInvalid(_receipt.status);
+        }
+
+        // If the transaction reverted, don't do anything.
+        if (_receipt.status == TransactionStatus.Reverted) {
+            emit TransactionReverted(_receipt.onchainTxId, _receipt.variant, _receipt.action);
+            return;
+        }
+
+        if (_receipt.variant == TransactionVariant.Withdraw) {
+            Withdraw memory withdraw = abi.decode(_receipt.action, (Withdraw));
+            IERC20(prove).safeTransfer(withdraw.account, withdraw.amount);
+        } else {
+            revert TransactionVariantInvalid();
+        }
     }
```

Comment:
since we're transferring the $PROVE directly now on withdraw, `claimableWithdrawal` is unused. Just remove the env var and the function from the interface